art/attacks/evasion/pixel_threshold.py

```diff
@@ -148,35 +172,30 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, max_iter: int
         if self.th is None:
             logger.info("Performing minimal perturbation Attack.")
 
-        if np.max(x) <= 1:
-            scale_input = True
-        else:
-            scale_input = False
+        
+        scale_input = False if np.max(x) > 1 else True
 
         if scale_input:
-            x = x * 255.0
+            x = self.depreprocess_images(x)
```

Comment:
This step of de-preprocessing should not be required in the attack's code. The input provided to `generate` is already expected to be un-preprocessed. Scaling with the `clip_values` (e.g. 255) as before is ok if the attack code only works for a different pixel range.

Reply:
If preprocessing is required for the model this preprocessing should be provided directly to the estimator under attack.